src/azure-cli-core/azure/cli/core/commands/cache.py

```diff
@@ -0,0 +1,131 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import timeit
+
+from knack.log import get_logger
+from knack.config import get_config_parser
+
+from azure.cli.core.extension import get_extensions
+from azure.cli.core._environment import get_config_dir
+
+logger = get_logger(__name__)
+
+
+CACHE_CONFIG_FILE = os.path.join(get_config_dir(), 'command_module_index.config')
+
+VERSION_STR = "VERSION"
+EXTENSIONS_STR = "EXTENSIONS"
+EXTENSIONS_LIST = ["{}-{}".format(ext.name, ext.version) for ext in get_extensions()]
+INDEX_STR = "INDEX"
+
+
+# return true if the cache exists and caching is enabled.
+def cache_exists():
+    use_cache = os.path.isfile(CACHE_CONFIG_FILE) and os.getenv('AZ_USE_CACHE', 'False').lower() == 'true'
+    logger.debug('use_cache is %s', use_cache)
+    return use_cache
```

Comment:
I don't think this method is necessary, but either way, if the cache exists it should return true whether the user wants to use it or not.